File: fsharp-giraffe-server/traveltimeplatform/src/model/RequestTimeFilterFast.fs
namespace traveltimeplatform.Model

open System
open System.Collections.Generic
open traveltimeplatform.Model.RequestLocation
open traveltimeplatform.Model.RequestTimeFilterFastArrivalSearches

module RequestTimeFilterFast = 

  //#region RequestTimeFilterFast


  type RequestTimeFilterFast = {
    Locations : RequestLocation[];
    ArrivalSearches : RequestTimeFilterFastArrivalSearches;
  }
  //#endregion
  

File: kotlin-spring/src/main/kotlin/org/openapitools/model/RequestTimeFilterFast.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import org.openapitools.model.RequestLocation
import org.openapitools.model.RequestTimeFilterFastArrivalSearches
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size
import javax.validation.Valid

/**
 * 
 * @param locations 
 * @param arrivalSearches 
 */
data class RequestTimeFilterFast(

    @field:Valid
    @get:Size(min=1)
    @field:JsonProperty("locations", required = true) val locations: kotlin.collections.List<RequestLocation>,

    @field:Valid
    @field:JsonProperty("arrival_searches", required = true) val arrivalSearches: RequestTimeFilterFastArrivalSearches
) {

}

